src/lib.rs

```diff
@@ -1,23 +1,87 @@
-use std::error::Error;
 use clap::Parser;
+use futures::try_join;
+use reqwest::{Client, Response};
+use serde_json::{json, Value};
+use std::error::Error;
+use std::process;
 
 #[derive(Parser, Debug)]
 pub struct Args {
-    #[arg(short,long, help("Message to encrypt"))]
-    pub encrypt: Option<String>, 
-    #[arg(short,long, help("Link to decrypt message from"))]
+    #[arg(short, long, help("Message to encrypt"))]
+    pub encrypt: Option<String>,
+
+    #[arg(
+        short,
+        long,
+        help("Time of expiry after encrypting a message"),
+        default_value_t = 50
+    )]
+    pub time_to_live: u8,
+
+    #[arg(short, long, help("Room id to decrypt message from"))]
     pub decrypt: Option<String>,
 }
 
-pub fn run(args: Args) -> Result<(), Box<dyn Error>> {
-   if let Some(message) = args.encrypt {
-       println!("Encrypting {message}");
-   }
+async fn handle_response(response: Response) -> Result<String, Box<dyn Error>> {
+    if response.status().is_success() {
+        Ok(response.text().await?)
+    } else {
+        Ok(format!(
+            "Request failed with exit code: {}",
+            response.status()
+        ))
+    }
+}
+
+async fn get_secret(client: &Client, room_id: String) -> Result<String, Box<dyn Error>> {
+    let response = (*client)
+        .get(format!("http://0.0.0.0:4000/api/secrets/{}", room_id))
+        .send()
+        .await?;
+    handle_response(response).await
+}
+
+async fn submit_secret(
+    client: &Client,
+    message: String,
+    expiry: u8,
+) -> Result<String, Box<dyn Error>> {
+    let secret: Value = json!({
+        "message": message,
+        "expiry": expiry,
+    });
+
+    let secret_string = secret.to_string();
+
```

Comment:
You are creating a JSON object just to immediately convert that to a String. Rust has a neat feature that allows us to define structs, that with the help of the [`serde` crate](https://crates.io/crates/serde), can represent JSON objects. We can use this to specify endpoints parameters or even to match the endpoint response. Note the `Serialize, Deserialize` in the `derive` macro come from the `serde` crate so you need to add that to the project in order to use it

```rust
#[derive(Debug, Serialize, Deserialize)]
struct CreateSecret {
    secret: String,
    duration: u64,
}
```

With this struct, instead of passing the `secret_string` to the  `.body` function and having to specify the `.header` to be JSON, we can do something like this:

```rust
let params = CreateSecret { secret, duration };

let url = format!("{}/api/secrets", API_URL);

let room: Room = self
    .client
    .post(url)
    .json(&params)
    .send()
    .await?
    .json()
    .await?;

Ok(room.room_id)
```

The `Room` type is also a struct that has the same `derive` macro and only has one element, `room_id`. We need to be explicit here with the type because Rust can't infer the result of an API call